### PR TITLE
chore(go): Make PK order be breaking

### PR DESCRIPTION
**Pull request by candiduslynx**

### CODE REVIEW

File: scripts/table_diff/changes/changes_test.go [hunk 1]
```diff
@@ -67,52 +87,40 @@ func Test_getChanges(t *testing.T) {
 					Breaking: true,
 				},
 				{
-					Text:     "Table `azure_appservice_functions`: column added with name `site_id` and type `String`",
-					Breaking: false,
+					Text: "Table `azure_appservice_functions`: column added with name `site_id` and type `String`",
```
Comment (erezrokah) [inline scripts/table_diff/changes/changes_test.go:90]: Let's not remove the setting of `Breaking` to `false`, I think it's better to be explicit with the test data, otherwise someone reading the test will have to go to the struct definition to understand what is being tested.
Also it's unrelated to the change in this PR and it makes it hard to review the new test

File: scripts/table_diff/changes/testdata/pr_6012_diff.txt [hunk 2]
```diff
@@ -0,0 +1,1042 @@
+diff --git a/plugins/source/github/codegen/recipes/actions.go b/plugins/source/github/codegen/recipes/actions.go
```
Comment (erezrokah) [inline scripts/table_diff/changes/testdata/pr_6012_diff.txt:1]: We can keep only the diff related to the `.md` files to make it easier to understand this file

File: scripts/table_diff/changes/changes.go [hunk 3]
```diff
@@ -66,40 +89,33 @@ func getColumnChanges(file *gitdiff.File, table string) (changes []change) {
...
-			}
-		} else {
+		addedColumn, ok := addedColumns[deletedName]
+		switch {
+		case !ok:
 			changes = append(changes, change{
 				Text:     fmt.Sprintf("Table %s: column removed %s from table", backtickStrings(table, deletedName)...),
 				Breaking: true,
 			})
+		case addedColumn.dataType != deletedColumn.dataType:
```
Comment (erezrokah) [inline scripts/table_diff/changes/changes.go:99]: I don't think this covers the same cases as before, for example what if `dataType` changes and a PK is added on the same column?
We will only report the first change instead of both, WDYT?

Also, this change seems unrelated to the purpose of this PR, so probably better to separate it as it makes harder to review the PR

File: scripts/table_diff/changes/changes.go [hunk 4]
```diff
@@ -47,13 +48,35 @@ func parseColumnChange(line string) (name string, dataType string, pk bool) {
 	return cleanName, result["dataType"], result["name"] != cleanName
 }
 
+func parsePKChange(line string) (names []string) {
+	for _, match := range pkRegex.FindAllStringSubmatch(line, -1) {
```
Comment (erezrokah) [inline scripts/table_diff/changes/changes.go:52]: Are we expecting to match the regex multiple times in a single line? If not can we use `FindStringSubmatch`?

File: scripts/table_diff/changes/changes.go [hunk 5]
```diff
@@ -47,13 +48,35 @@ func parseColumnChange(line string) (name string, dataType string, pk bool) {
...
+
 func getColumnChanges(file *gitdiff.File, table string) (changes []change) {
 	addedColumns := make(map[string]column)
 	deletedColumns := make(map[string]column)
+	var addedPK, deletedPK []string
 	for _, fragment := range file.TextFragments {
 		for _, line := range fragment.Lines {
 			name, dataType, pk := parseColumnChange(line.Line)
 			if name == "" || dataType == "" {
+				pkChanges := parsePKChange(line.Line)
```
Comment (erezrokah) [inline scripts/table_diff/changes/changes.go:71]: Can't we move this up just at the beginning of the loop here:
https://github.com/cloudquery/cloudquery/pull/6021/files#diff-6910d6ee0a1eadb1d8f2122f804f6ab3ef6e4c6d040b631527d79e73b4678c50R69
to avoid the nesting?

**Review by erezrokah — CHANGES_REQUESTED** [review]
Thanks for the PR @candiduslynx 🚀 

I think the approach in this PR will work well. Added a few comments

File: scripts/table_diff/changes/changes.go (hunk 4 above)
Comment (candiduslynx) [reply scripts/table_diff/changes/changes.go:52]: yes, sure!

**Review by erezrokah — APPROVED** [review]
🚀

**Comment by candiduslynx** [comment]
Need to be revisited after discussion at #6043 is complete
